src/main/java/org/apache/commons/lang3/exception/ExceptionUtils.java

```diff
@@ -586,6 +586,30 @@ public static int indexOfType(final Throwable throwable, final Class<? extends T
         return indexOf(throwable, type, fromIndex, true);
     }
 
+    /**
+     * Return whether the given throwable is a checked exception:
+     * that is, neither a RuntimeException nor an Error.
+     *
+     * <pre>
+     * ExceptionUtils.isCheckedException(new Exception())                    = true
+     * ExceptionUtils.isCheckedException(new SQLException())                 = true
+     * ExceptionUtils.isCheckedException(new RuntimeException())             = false
+     * ExceptionUtils.isCheckedException(new IllegalArgumentException(""))   = false
+     * ExceptionUtils.isCheckedException(new Throwable())                    = true
+     * ExceptionUtils.isCheckedException(null)                               = false
+     * </pre>
+     *
+     * @param throwable the throwable to check
+     * @return whether the throwable is a checked exception
+     * @see java.lang.Exception
+     * @see java.lang.RuntimeException
+     * @see java.lang.Error
+     * @since 3.12.1
+     */
+    public static boolean isCheckedException(final Throwable throwable) {
+        return throwable != null && !(throwable instanceof RuntimeException || throwable instanceof Error);
```

Comment:
`instanceof` handles null. Does this work if you remove the `throwable != null &&`?

I would expect in the majority of cases this would be called with a non-null object so this simplifies the code.

Reply:
I prefer to leave all options open. it is more versatile. ...

Reply:
My point was that you do not need the `throwable != null`. It should still pass your unit tests (i.e. return false if null is passed).

Reply:
No. Return true without the null check